### PR TITLE
Make request and response HttpClient buffers configurable

### PR DESCRIPTION
**Pull request by sopel39**
When using GSS requests might exceed request buffer size.
Therefore those buffers needs to be configurable.

### CODE REVIEW

File: http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java [hunk 1]
```diff
@@ -218,6 +220,34 @@ public HttpClientConfig setMaxContentLength(DataSize maxContentLength)
...
     }
 
+    @NotNull
+    public DataSize getRequestBufferSize()
+    {
+        return requestBufferSize;
+    }
+
+    @Config("http-client.request-buffer-size")
+    @MaxDataSize("32MB")
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java]: Bean validations need to go on the getter (the validation framework validates the entire object after it is built)

File: http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java [hunk 2]
```diff
@@ -218,6 +220,34 @@ public HttpClientConfig setMaxContentLength(DataSize maxContentLength)
...
+    }
+
+    @NotNull
+    public DataSize getResponseBufferSize()
+    {
+        return responseBufferSize;
+    }
+
+    @Config("http-client.response-buffer-size")
+    @MaxDataSize("32MB")
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java]: Same here

File: http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java [hunk 3]
```diff
@@ -200,6 +200,10 @@ public JettyHttpClient(
 
         httpClient = new AuthorizationPreservingHttpClient(transport, sslContextFactory);
 
+        // request and response buffer size
+        httpClient.setRequestBufferSize((int) config.getRequestBufferSize().toBytes());
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java]: Use `toIntExact` instead of a cast

**Comment by sopel39** [comment]
ac

**Pull request merged**